jetson/nav/rover.hpp

```diff
@@ -63,178 +65,118 @@ enum class NavState
 
-    DriveStatus drive( const double distance, const double bearing, const bool target = false );
+    int getLeftMisses() const;
 
-    void drive( const int direction, const double bearing );
+    int getRightMisses() const;
 
-    bool turn( Odometry& destination );
+    int getLeftHits() const;
 
-    bool turn( double bearing );
+    int getRightHits() const;
 
-    void stop();
+    void setAutonState(AutonState state);
 
-    bool updateRover( RoverStatus newRoverStatus );
+    void setOdometry(Odometry const& odometry);
 
-    RoverStatus& roverStatus();
+    void updateTargets(shared_ptr<Environment> const& env, shared_ptr<CourseProgress> const& course);
 
-    PidLoop& distancePid();
-
-    PidLoop& bearingPid();
+    void setState(NavState state);
 
-    const double longMeterInMinutes() const;
+    void resetMisses();
 
-  
 private:
     /*************************************************************************/
     /* Private Member Functions */
     /*************************************************************************/
-    void publishJoystick( const double forwardBack, const double leftRight, const bool kill );
+    void publishAutonDriveCmd( const double leftVel, const double rightVel);
 
-    bool isEqual( const Obstacle& obstacle1, const Obstacle& obstacle2 ) const;
-
-    bool isEqual( const Odometry& odometry1, const Odometry& odometry2 ) const;
-
-    bool isEqual( const Target& target, const Target& target2 ) const;
-
-    bool isTurningAroundObstacle( const NavState currentState ) const;
+    static bool isTurningAroundObstacle(NavState currentState);
 
     /*************************************************************************/
     /* Private Member Variables */
     /*************************************************************************/
 
-    // The rover's current status.
-    RoverStatus mRoverStatus;
-
     // A reference to the configuration file.
     const rapidjson::Document& mRoverConfig;
 
     // A reference to the lcm object that will be used for
     // communicating with the actual rover and the base station.
     lcm::LCM& mLcmObject;
 
-    // The pid loop for driving.
-    PidLoop mDistancePid;
-
     // The pid loop for turning.
     PidLoop mBearingPid;
 
-  
-
     // The conversion factor from arcminutes to meters. This is based
     // on the rover's current latitude.
     double mLongMeterInMinutes;
-};
 
-#endif // ROVER_HPP
+    // The rover's current navigation state.
+    NavState mCurrentState{NavState::Off};
+
+    // The rover's current auton state.
+    AutonState mAutonState{};
+
+    // The rover's current path. The path is initially the same as
+    // the rover's course, however, as waypoints are visited, the
+    // are removed from the path but not the course.
+
+    // The rover's current odometry information.
+    Odometry mOdometry{};
+
+    // The rover's current target information from computer
+    // vision.
+    Target mTargetLeft{-1.0, 0.0, 0};
```

Comment:
is -1.0, 0.0, 0 the default? can this just be the default initialization for Target?

Reply:
It's an lcm auto generated, is that possible?

Reply:
wym lcm auto generated?

Reply:
Target.hpp is auto generated by LCM, so I can't add a ctor I don't think that would set  it to -1.0, 0.0, 0